LibThreatClassic2.lua

```diff
@@ -1727,6 +1727,31 @@ function ThreatLib:UnitDetailedThreatSituation(unit, target)
 	return isTanking, threatStatus, threatPercent, rawThreatPercent, floor(threatValue)
 end
 
+------------------------------------------------------------------------
+-- :UnitThreatPercentageOfLead("unit", "mob")
+-- Arguments: 
+--  string - unitID of the unit to get threat information for.
+--  string - unitID of the target unit to reference.
+-- Returns:
+--  integer - returns the relative threat percentage compared to the next highest (if tanking) or highest on the threat list
+------------------------------------------------------------------------
+function ThreatLib:UnitThreatPercentageOfLead(unit, target)
+	local unitGUID, targetGUID = UnitGUID(unit), UnitGUID(target)
+	if not (unitGUID and targetGUID) then return nil end
+
+	local unitValue = self:GetThreat(unitGUID, targetGUID)
+	if unitValue == 0 then return 0 end
+
+	local maxValue = 0
+	for otherGUID in next, threatTargets do
```

Comment:
I really don't like this syntax. `next, table` is a really ugly way of providing an iterator.

can you please change it to

`for otherGUID in pairs(threatTargets) do`

i will merge this then. Had some time to check the provided links and this function won't hurt.

Reply:
It's a slight performance improvement if you just want the keys. I switched it to the more standard format anyway. For future reference, `next()` literally is the iterator, followed by a persistent value and an initial value. `pairs()` and `ipairs()` are simply wrappers for this more complex syntax in case you're interested in making your own iterators in future projects.